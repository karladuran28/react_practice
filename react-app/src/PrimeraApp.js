import React from 'react';
import PropTypes from 'prop-types';
/* functional component */

const PrimeraApp = ({ saludo, subtitulo }) => {

    return (
        <>
            <h1>Hola Mundo</h1>
            <h2>Mi primera aplicaci√≥n</h2>
            <h3>{ saludo }</h3>
            <p>{ subtitulo }</p> 
        </>
    );

};

/* tipo de dato de cada prop */
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

/* Valores default */
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo',
}

export default PrimeraApp;