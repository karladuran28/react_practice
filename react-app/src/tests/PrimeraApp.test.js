import React from 'react';
import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme'

describe('Pruebas en <PrimeraApp />', () => {

/*     test('demostrar mostrar Hola soy Karla', () => {
        const saludo = 'Hola soy Karla';
        const { getByText } = render( <PrimeraApp saludo={ saludo }/> );

        expect(  getByText(saludo) ).toBeInTheDocument();

    }); */

    test('debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola soy Karla';
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/> );
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola soy Karla';
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo }
                subtitulo={ subtitulo }
            /> 
        );

        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);
        
        expect( textoParrafo ).toBe( subtitulo );


    });
    
    
    
});