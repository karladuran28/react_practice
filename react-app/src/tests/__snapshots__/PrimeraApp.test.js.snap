// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <h2>
    Mi primera aplicaci√≥n
  </h2>
  <h3>
    Hola soy Karla
  </h3>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
